### PR TITLE
Allow Viral Tasks

### PR DESCRIPTION
**Pull request by Spedwards**
Now to add a viral task, in addition to previous method, call `Task.addTasks(youTask)`.

### CODE REVIEW

File: task.js [hunk 1]
```diff
@@ -1,42 +1,21 @@
 let mod = {};
 module.exports = mod;
+mod.tasks = [];
+mod.populate = function() {
+    Task.addTasks(Task.guard, Task.defense, Task.claim, Task.reserve, Task.mining, Task.pioneer, Task.attackController, Task.robbing, Task.reputation);
```
Comment (karlthepagan) [inline task.js]: one line per task for easier merges

also please alphabetize the task names

Comment (Spedwards) [inline task.js]: Sure thing.

**Comment by Dasku** [comment]
yey!!!

File: task.js [hunk 2]
```diff
@@ -2,15 +2,17 @@ let mod = {};
...
-    Task.addTasks(Task.attackController);
-    Task.addTasks(Task.claim);
-    Task.addTasks(Task.defense);
-    Task.addTasks(Task.guard);
-    Task.addTasks(Task.mining);
-    Task.addTasks(Task.pioneer);
-    Task.addTasks(Task.reputation);
-    Task.addTasks(Task.reserve);
-    Task.addTasks(Task.robbing);
+    Task.addTasks(...[
```
Comment (Spedwards) [inline task.js]: An array so you can have a following comma on the last item, making for easier merges.

File: task.js [hunk 3]
```diff
@@ -1,42 +1,31 @@
...
-    }
+        Task.reserve,
+        Task.robbing,
+    ]);
+};
+mod.addTasks = (...task) => Task.tasks.push(...task);
+// load task memory & flush caches
+mod.flush = function () {
+    Task.tasks.forEach(task => {
+        if (task.flush) task.flush();
```
Comment (chobobobo) [inline task.js]: Do we want to make sure this is actually a function as well while we're at it?

File: task.js [hunk 4]
```diff
@@ -1,42 +1,31 @@
...
-        Task.robbing,
-        Task.reputation,
-    ];
-    for (let i = tasks.length - 1; i >= 0; i--) {
-        if (tasks[i].register) {
-            tasks[i].register();
-        }
-    }
+    Task.tasks.forEach(task => {
+        if (task.register) task.register();
```
Comment (chobobobo) [inline task.js]: as above.

File: task.js (hunk 3 above)
Comment (karlthepagan) [inline task.js]: I'm ambivalent about strictly checking the type like that.

File: task.js (hunk 2 above)
Comment (karlthepagan) [inline task.js]: :+1:

**Pull request merged**